Battery modeling updates

emmasmit5 opened the pull request:
Corrects errors in Model.m and separates it to two discharging and charging models. SOH degradation model is included but default is no degradation effects.

yipengliu1208 committed:
Update 1

 battery pack

yipengliu1208 committed:
Update 2

default of battery settings

yipengliu1208 committed:
Update 3

Cost models

yipengliu1208 committed:
Update 4

charging and degradation models

yipengliu1208 committed:
Delete Model.m

No long being used

emmasmit5 committed:
Merge branch 'main' into Batt_Merge_PAO

emmasmit5 committed:
change degredation toggle to be in battery specs not settings

yipengliu1208 committed:
Update the Testmodels

The "TestGroundCharging" model is not available since we chanced the groundcharging model a lot. So the previous hardcoded reference values are no longer correct.

prmoko reviewed (changes requested):
@emmasmit5 and @yipengliu1208, please see the attached document for all feedback that must be addressed. There are significant formatting changes that need to be made. Additionally, some of the functions do not run correctly and return errors.

Please leave a comment once you fix the changes. Then, I will review and test the code again.


[Feedback-PR72.docx](https://github.com/user-attachments/files/22568967/Feedback-PR72.docx)

yipengliu1208 committed:
Update BattAgingOffDesign.m

yipengliu1208 committed:
Update from Notes

emmasmit5 committed:
clean up

emmasmit5 committed:
Delete TestGroundCharge.m

emmasmit5 committed:
HEA sizing updates

fix compatibility issues between HEA battery sizing

emmasmit5 committed:
Merge branch 'main' into Batt_Merge_PAO

emmasmit5 committed:
Merge branch 'main' into Batt_Merge_PAO

emmasmit5 committed:
updates from main and fix battery sizing work

prmoko committed:
Updated CHANGELOG

Corrected the version number on the code release.

prmoko committed:
Minor Formatting Changes

Made a few more minor formatting changes to ensure that the code can be released cleanly.

prmoko reviewed (changes requested):
Thanks for making the changes, @emmasmit5!

I have a couple of questions about the changes made in PropAnalysis, PowerAvailable, and EvalClimb. Will schedule a meeting to review the changes.

prmoko committed:
Updated README

Added the CostPkg and updated other letter labels accordingly.

prmoko committed:
Updated EvalClimb

Added RecomputeSplits back into the code for consistency. Will be updated in a future release to accommodate configurations beyond PHE.

prmoko committed:
Updated PowerAvailable

Removed the electric motor power output code. The result obtained here should match the SLS power available earlier in the code (Line 93).

prmoko committed:
Updated README

Updated the README with the correct release date and updated the publication to the GPSA Framework.

prmoko committed:
Updated CHANGELOG

Updated the CHANGELOG with the correct release date.

prmoko reviewed (approved):
Thanks for making all the changes and meeting yesterday, @emmasmit5. All changes look good now and will be merged to the main branch.
